packages/core/modules-sdk/src/medusa-app.ts

```diff
@@ -417,55 +442,73 @@ async function MedusaApp_({
         await MedusaModule.migrateDown(
           moduleResolution.definition.key,
           moduleResolution.resolutionPath as string,
+          sharedContainer,
           moduleResolution.options,
           moduleResolution.moduleExports
         )
       } else {
         await MedusaModule.migrateUp(
           moduleResolution.definition.key,
           moduleResolution.resolutionPath as string,
+          sharedContainer,
           moduleResolution.options,
           moduleResolution.moduleExports
         )
       }
     }
+  }
 
-    const linkModuleOpt = { ...(linkModuleOptions ?? {}) }
-    linkModuleOpt.database ??= {
-      ...(sharedResourcesConfig?.database ?? {}),
-    }
+  const runMigrations: RunMigrationFn = async (): Promise<void> => {
+    await applyMigration({
+      modulesNames: Object.keys(allModules),
+    })
+
+    const options: Partial<ModuleServiceInitializeOptions> =
+      "scope" in linkModuleOrOptions
+        ? { ...linkModuleOrOptions.options }
+        : {
+            ...(linkModuleOrOptions as Partial<ModuleServiceInitializeOptions>),
+          }
 
-    if (revert) {
-      revertLinkModuleMigration &&
-        (await revertLinkModuleMigration(
-          {
-            options: linkModuleOpt,
-            injectedDependencies,
-          },
-          linkModules
-        ))
-    } else {
-      linkModuleMigration &&
-        (await linkModuleMigration(
-          {
-            options: linkModuleOpt,
-            injectedDependencies,
-          },
-          linkModules
-        ))
+    options.database ??= {
+      ...sharedResourcesConfig?.database,
     }
-  }
 
-  const runMigrations: RunMigrationFn = async (
-    linkModuleOptions
-  ): Promise<void> => {
-    await applyMigration(linkModuleOptions)
+    await linkModuleMigration(
+      {
+        options,
+        injectedDependencies,
+      },
+      linkModules
+    )
   }
 
-  const revertMigrations: RunMigrationFn = async (
-    linkModuleOptions
+  const revertMigrations: RevertMigrationFn = async (
+    modulesNames
   ): Promise<void> => {
-    await applyMigration(linkModuleOptions, true)
+    await applyMigration({
+      modulesNames,
+      revert: true,
+    })
+
+    const options: Partial<ModuleServiceInitializeOptions> =
+      "scope" in linkModuleOrOptions
+        ? { ...linkModuleOrOptions.options }
+        : {
+            ...(linkModuleOrOptions as Partial<ModuleServiceInitializeOptions>),
+          }
+
+    options.database ??= {
+      ...sharedResourcesConfig?.database,
+    }
+
+    await revertLinkModuleMigration(
```

Comment:
Q: I suppose we will move that in a later pr right?

Reply:
Yeah. First have to work on a `sync links` implementation and then remove it.